fedimint-api/src/db/mod.rs

```diff
@@ -336,15 +409,25 @@ impl<'isolated, 'parent, T: Send + Encodable + 'isolated> IDatabaseTransaction<'
     }
 }
 
+impl<'isolated, 'parent: 'isolated, T: Send + Encodable + 'isolated> NotifyingTransaction<'isolated>
+    for IsolatedDatabaseTransaction<'isolated, 'parent, T>
+{
+    fn add_notification_key(&mut self, key: &[u8]) {
```

Comment:
I see. I was confused what was the solution for `IsolatedDatabaseTransaction without modifyng `IDatabaseTransaction`.

I think it works, but has a problem of composing all that well - now we'll have to `NotifyingTransaction` for every `IsolatedDatabaseTransaction`-like thing we need? Which for our purposes might be Ok ... I guess.

Please make sure this all works across different levels:

* one side is working on a `IsolatedDatabase` and waits for stuff that is being inserted by `Database` (just with the same prexix)
* the opposite: one side waits on `Database`, while the write is being done from `IsolatedDatabaseTransaction` level
* both isolated and both non-isolated.

If all works, then good enough I guess. :)